swingset-demo/src/main/java/com/nqadmin/swingset/demo/Example4UsingHelper.java

```diff
@@ -245,10 +245,12 @@ protected void populateSSComboBoxes() {
 	}
 
 	/**
-	 * Obtain and set the PK value for the new record.
+	 * Obtain and return the primary key value for the new record.
 	 */
 	@Override
-	protected void retrieveAndSetNewPrimaryKey() {
+	protected String retrieveNewPrimaryKey() {
```

Comment:
This seemed like a cleaner way to handle in practice.

Reply:
This is beyond me.

Reply:
When you add a new record to the database, the user can populate the primary key column if there is one or you can rely on a unique index provided by the database. Previously in my form helper class I had a abstract method to say "take care of the primary key for the new record". In the form helper, I actually maintain an SSTextField bound to the primary key column so I thought it would be better to change the API to just request the new value and let the helper take care of setting it. Eliminates a little code and hopefully makes the helper more helpful.

Reply:
The idea is to query the database for the next value in the appropriate sequence (part_data_seq for our part_data table) and return the result.
`executeQuery("SELECT nextval('part_data_seq') as nextVal;");`